android/src/main/java/com/thunderhead/one_sdk_flutter/OnePlugin.java

```diff
@@ -29,132 +40,271 @@
-    String apiKey = (String) arguments.get("apiKey");
-    String sharedSecret = (String) arguments.get("sharedSecret");
-    String userId = (String) arguments.get("userID");
-    String host = (String) arguments.get("host");
-    boolean adminMode = (Boolean) arguments.get("adminMode");
-
-    final OneConfiguration oneConfiguration = new OneConfiguration.Builder()
-            .siteKey(siteKey)
-            .apiKey(apiKey)
-            .sharedSecret(sharedSecret)
-            .userId(userId)
-            .host(URI.create(host))
-            .touchpoint(URI.create(touchpointURI))
-            .mode(adminMode ? OneModes.ADMIN_MODE : OneModes.USER_MODE)
-            .build();
-    One.setConfiguration(oneConfiguration);
-    result.success(true);
-  }
-
-  private void setLogLevel(MethodCall call, final Result result) {
-    OneLoggingConfiguration oneLoggingConfiguration;
-    final Map<String, Boolean> arguments = call.arguments();
-    Boolean enabled = arguments.get("logLevel");
-
-    if (enabled == null) {
-      enabled = true;
+
+    private void sendInteraction(String interactionPath, HashMap properties, final Result result) {
+        final OneRequest sendInteractionRequest = new OneRequest.Builder()
+                .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
+                .properties(properties)
+                .build();
+        executor.submit(() -> {
+            OneResponse response;
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                try {
+                    response = One.sendInteraction(sendInteractionRequest).join();
+                    One.processResponse(response);
+                    HashMap<String, Object> responseMap = responseObjectToHashMap(response);
+                    result.success(responseMap);
+                } catch (ExecutionException error) {
+                    result.error(Integer.toString(error.hashCode()), error.getLocalizedMessage(), error);
+                    Log.e(LOG_TAG, "[Thunderhead] Completion Error: " + error.getCause());
+                } catch (OneSDKError error) {
+                    result.error(Integer.toString(error.getSystemCode()), error.getLocalizedMessage(), error);
+                    Log.e(LOG_TAG, "[Thunderhead] SDK Error: " + error.getErrorMessage());
+                } catch (OneAPIError error) {
+                    result.error(Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage(), error);
+                    Log.e(LOG_TAG, "[Thunderhead] Api Error: " + error.getErrorMessage());
+                }
+            } else {
+                try {
+                    response = One.sendInteractionLegacySupport(sendInteractionRequest).join();
+                    One.processResponse(response);
+                    HashMap<String, Object> responseMap = responseObjectToHashMap(response);
+                    result.success(responseMap);
+                } catch (ExecutionException error) {
```

Comment:
You're missing the other errors.  This can be simplified:

```
try {
    OneResponse response;
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
        response = One.sendInteraction(true, sendInteractionRequest).join();
    } else {
        response = One.sendInteractionLegacySupport(true, sendInteractionRequest).join();
    }

    One.processResponse(response);
    HashMap<String, Object> responseMap = responseObjectToHashMap(response);
    result.success(responseMap);
} catch (ExecutionException error) {
    result.error(Integer.toString(error.hashCode()), error.getLocalizedMessage(), error);
    Log.e(LOG_TAG, "[Thunderhead] Completion Error: " + error.getCause());
} catch (OneSDKError error) {
    result.error(Integer.toString(error.getSystemCode()), error.getLocalizedMessage(), error);
    Log.e(LOG_TAG, "[Thunderhead] SDK Error: " + error.getErrorMessage());
} catch (OneAPIError error) {
    result.error(Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage(), error);
    Log.e(LOG_TAG, "[Thunderhead] Api Error: " + error.getErrorMessage());
}

Reply:
Also you are missing throw errors.  I edited my above comment to add it.